pkg/domain/domain.go

```diff
@@ -1067,7 +1067,15 @@ func (do *Domain) InitDistTaskLoop() error {
 		}
 	})
 
-	taskManager := storage.NewTaskManager(do.sysSessionPool)
+	distTaskSessPool := do.sysSessionPool
+	if keyspace.IsRunningOnUser() {
+		sp, err := do.GetKSSessPool(keyspace.System)
+		if err != nil {
+			return err
+		}
+		distTaskSessPool = sp
+	}
+	taskManager := storage.NewTaskManager(distTaskSessPool)
```

Comment:
this cannot be merged now, current task still runs on its owner keyspace, we can merge it when we are ready to switch to DXF service

Reply:
What else do we need to do? I can run add-index job normally after this PR.

Reply:
we shouldn't change this for all places, and only change where add-index uses `TaskManager`

Reply:
For a classical kernel, it should use advancenSessionPool